btree/btree.c

```diff
@@ -2966,7 +2971,7 @@ static void level_cleanup(struct m0_btree_oimpl *oi, struct m0_be_tx *tx)
 	 * already taken by this thread.
 	 */
 	int i;
-	for (i = 0; i <= oi->i_used; ++i) {
+	for (i = 0; i < oi->i_height; ++i) {
```

Comment:
Why did we move from i_used to i_height ?

Reply:
So, it might be a corner case scenario.
Consider the scenario where at first traversal we find the height of tree 4 and at worst case allocated new nodes at all the levels from 0 to 3 level. Now in P_CHECK, we found that path_check function fails but height is same. Now, we start traversing again from P_DOWN , now consider height of tree decreased, now height is 3 i.e. our oi->i_used will be maximum 2. so, consider we havn't used any of the allocated node allocated at the time of 1st traversal ,  so ideally we should free all the allocated node from level 0 to 3 .But for second traversal oi->i_used will be 2.
 Therefore , we should used oi->i_height at time of node_free instead of oi->i_used.